File: providers/anthropic.js
module.exports = {
  
  name: 'anthropic',
  aliases: ['claude'],
  displayName: 'Anthropic',
  description: 'Anthropic Messages API — Claude models',
  baseUrl: 'https://api.anthropic.com/v1/messages',
  apiMode: 'anthropic_messages',
  envVars: ['ANTHROPIC_API_KEY', 'ANTHROPIC_BASE_URL'],
  authType: 'api_key',
  defaultModel: 'claude-sonnet-4-6',
  defaultMaxTokens: 8192,
  signupUrl: 'https://console.anthropic.com',
  supportsHealthCheck: true,
  defaultHeaders: { 'anthropic-version': '2023-06-01' },
  modelProfiles: {
    'claude-opus-4': { supportsReasoning: true, supportsThinking: true, combinedBudget: false, contextWindow: 200000, maxOutputTokens: 8192 },
    'claude-sonnet-4': { supportsReasoning: true, supportsThinking: true, combinedBudget: false, contextWindow: 200000, maxOutputTokens: 8192 },
    'claude-3': { supportsReasoning: false, supportsThinking: true, combinedBudget: false, contextWindow: 200000, maxOutputTokens: 4096 },
    '*': { supportsReasoning: false, supportsThinking: false, combinedBudget: false, contextWindow: 200000, maxOutputTokens: 4096 },
  },

};

File: backend-py/app/providers/anthropic.py
"""
Provider config for Anthropic.
"""
from __future__ import annotations
INFO: dict[str, object] = {'name': 'Anthropic', 'aliases': ['claude'], 'display_name': 'Anthropic', 'description': 'Anthropic Messages API — Claude models', 'base_url': 'https://api.anthropic.com/v1/messages', 'api_mode': 'anthropic_messages', 'env_vars': ['ANTHROPIC_API_KEY', 'ANTHROPIC_BASE_URL'], 'auth_type': 'api_key', 'default_model': 'claude-sonnet-4-6', 'default_max_tokens': 8192, 'default_headers': {'anthropic-version': '2023-06-01'}, 'signup_url': 'https://console.anthropic.com', 'supports_health_check': True, 'model_profiles': {'claude-opus-4': {'supportsReasoning': True, 'supportsThinking': True, 'combinedBudget': False, 'contextWindow': 200000, 'maxOutputTokens': 8192}, 'claude-sonnet-4': {'supportsReasoning': True, 'supportsThinking': True, 'combinedBudget': False, 'contextWindow': 200000, 'maxOutputTokens': 8192}, 'claude-3': {'supportsReasoning': False, 'supportsThinking': True, 'combinedBudget': False, 'contextWindow': 200000, 'maxOutputTokens': 4096}, '*': {'supportsReasoning': False, 'supportsThinking': False, 'combinedBudget': False, 'contextWindow': 200000, 'maxOutputTokens': 4096}}}

def resolveBaseUrl() -> str:
    return 'https://api.anthropic.com/v1/messages'

def resolveApiKey(envKey: str | None=None) -> str | None:
    return envKey